### PR TITLE
add `tar_load_all()`

### PR DESCRIPTION
**Pull request by malcolmbarrett**
# Prework

* [x] I understand and agree to the [code of conduct](https://ropensci.org/code-of-conduct/) and the [contributing guidelines](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [x] I have already submitted a [discussion topic](https://github.com/ropensci/targets/discussions) or [issue](https://github.com/ropensci/targets/issues) to discuss my idea with the maintainer.

# Related GitHub issues and pull requests

* Ref: #772

# Summary

Please explain the purpose and scope of your contribution.

This PR adds a small helper function, `tar_load_all()`, which loads all targets.

A point of discussion: I stuck with `tar_load_all()` like you suggested, but it occurs to me that a user of devtools might expect this function to do something slightly different e.g. load all functions that the pipeline is using, more similar to what `load_all()` does. Do you think that's an issue? `tar_load_everything()` is accurate but a bit long (if a non-issue thanks to autocompleting).

### CODE REVIEW

**Comment by wlandau** [comment]
Good point about the name, I like `tar_load_everything()` for the reasons you describe and its more obvious correspondence to `tar_load(everything())`.

File: R/tar_load.R [hunk 1]
```diff
@@ -50,3 +51,24 @@ tar_load <- function(
...
+#' @rdname tar_load
+tar_load_all <- function(
+  branches = NULL,
+  meta = tar_meta(targets_only = TRUE, store = store),
+  strict = TRUE,
+  silent = FALSE,
+  envir = parent.frame(),
+  store = targets::tar_config_get("store")
+) {
+  tar_load(
```
Comment (wlandau) [inline R/tar_load.R:65]: Would you include `force(envir)` here at the top? That way, `parent.frame()` refers to the parent frame of your function and not that of `tar_load()`.

**Review by wlandau — COMMENTED** [review]
Looks great! Would you add an entry for the function in `_pkgdown.yml`? After that, `force(envir)`, and the name change to `tar_load_everything()`, I will merge.

File: R/tar_load.R (hunk 1 above)
Comment (malcolmbarrett) [reply R/tar_load.R:65]: good point

**Comment by malcolmbarrett** [comment]
Done, thanks!

**Review by wlandau — APPROVED** [review]
Thanks, @malcolmbarrett!

**Pull request merged**